server.py

```diff
@@ -151,12 +153,20 @@ def get(key):
             os.rename(uri, path.join(settings['assetdir'], asset['asset_id']))
             uri = path.join(settings['assetdir'], asset['asset_id'])
 
+    if 'youtube_asset' in asset['mimetype']:
+        uri, asset['name'] = download_video_from_youtube(uri, asset['asset_id'])
+        asset['mimetype'] = 'video'
+        asset['is_processing'] = 1
+
     asset['uri'] = uri
 
     if "video" in asset['mimetype']:
-        video_duration = get_video_duration(asset['uri'])
-        if video_duration:
-            asset['duration'] = int(video_duration.total_seconds())
+        if asset['is_processing'] == 0:
+            video_duration = get_video_duration(uri)
```

Comment:
We can get information about the duration of the video from youtube_dl, but it does not make much sense: after downloading a video from YouTube, we work with the asset as with a regular video asset (https://github.com/Screenly/screenly-ose/commit/c07d82319cff7b7bfbcafe607c871a378d98c456#diff-d355af669766ee3562643e8e5034f715R158). This means that when editing an asset, for example, changing the date - in any case, `get_video_duration` will be called.